content/en/docs/plugins/plugin-api-reference.md

```diff
@@ -130,7 +158,7 @@ When managing plugin-level state, keep the following in mind:
 * On failure, make sure to return a meaningful error message in the next call to `plugin_get_last_error()`.
 * On failure, the plugin framework will *not* call plugin_destroy, so do not return any allocated state.
 
-The format of the config string is entirely determined by the plugin author, and is passed unchanged from Falco/the application using the plugin framework to the plugin. Given that, semi-structured formats like JSON/YAML are preferable to free-form text.
+The format of the config string is entirely determined by the plugin author, and by default is passed unchanged from Falco/the application using the plugin framework to the plugin. However, semi-structured formats like JSON/YAML are preferable to free-form text. In those cases, the plugin author can provide a schema describing the config string contents by implementing the optional `get_init_schema` function. If so, the `init()` function can assume the passed-in configuration string to always be well-formed, and can avoid performing any error handling. The plugin framework will take care of automatically parsing it against the provided schema and generating ad-hoc errors accordingly. Please refer to the documentation of `get_init_schema` for more details.
```

Comment:
Shall we recommend JSON?

Reply:
I think it is pretty much recommended right now. By using JSON, you have access to the schema feature using JSON Schema, which delegates the configuration validation to the framework. I think this the `get_init_schema` doc section make this explicit, but if you want I can add few lines here too.